ProductsExplorer.php

```diff
@@ -1,97 +1,125 @@
+                            .$UserId. " AND ProductStatus = ".$Code;
+                }
+            }
+
+            # main work happens here
+            $DB->Query($SQL);
+            $Ret = array();
+            while ($Row = $DB->NextRow()) {
+                $Ret[] = $Row["ProductId"];
+            }
+            $DB->Close();
+            return $Ret;
+        }
+
+        /**
+        * Get the products with a certain product type defined in Product class.
+        * @param const $Type Product type constant.
+        * @return An array of product IDs of all the products of the required type.
+        */
+        public static function GetProductsByType($Type) {
+            $Code = Product::GetTypeCodeForDB($Type);
+            $Ret = array();
+            $DB = new Database();
+
+            $StaCode = Product::GetStatusCodeForDB(Product::PSTATUS_ACTIVE);
+            $SQL = "SELECT ProductId FROM Products WHERE ProductType = ".$Code." AND ProductStatus = ".$StaCode;
+            $DB->Query($SQL);
+            while ($Row = $DB->NextRow()) {
+                $Ret[] = $Row["ProductId"];
+            }
+            $DB->Close();
+            return $Ret;
+        }
+
+        /**
+        * Get all active products ordered by productId.
+        * @return An array of ALL product ID that are ACTIVE.
+        */
+        public static function GetAllActiveProducts() {
+            $Code = Product::GetStatusCodeForDB(Product::PSTATUS_ACTIVE);
+            $Ret = array();
+            $DB = new Database();
+            $DB->Query("SELECT ProductId FROM Products WHERE ProductStatus = "
+                    .$Code." ORDER BY ProductId ASC");
+            while ($Row = $DB->NextRow()) {
+                $Ret[] = $Row["ProductId"];
+            }
+            $DB->Close();
+            return $Ret;
+        }
+
+        /**
+         * Get the next latest products after the specified product in chronological order.
+         * @param $StartId Specifies a product from which to start, exclusive.
+         * @param $ListLength Determine how many products to be returned
+         * @return An array of latest products after the specified product.
+         * @TODO: right now products are selected based on ID rather than release date.
+         */
+        public static function GetNextProducts($StartId, $ListLength)
+        {
+            $Code = Product::GetStatusCodeForDB(Product::PSTATUS_ACTIVE);
+            $Ret = array();
+            $DB = new Database();
+
+            if ($StartId == -1) {
+                $DB->Query("SELECT ProductId FROM Products WHERE ProductStatus = "
+                    .$Code." ORDER BY ProductId DESC");
+            } else {
+                $DB->Query("SELECT ProductId FROM Products WHERE ProductStatus = "
+                    .$Code." AND ProductId < ".$StartId." ORDER BY ProductId DESC");
+            }
+            $Index = 0;
+            while($Row = $DB->NextRow() and $Index++ < $ListLength) {
```

Comment:
Could you use "&&" instead of "and" to keep it consistent (all other places use &&).